### DISCUSSION TITLE
Error while moving/renaming file from one virtual folder to another virtual folder mounted on same account

### DISCUSSION THREAD

**Discussion by rashmimaharana** [drakkan/sftpgo#946] [Q&A]
Hi Nicola,

We have hosted sftpgo on GCP running inside GKE cluster. We use GCS as backend for saving files. In one GCS bucket we have created 2 folders say folder_1 and folder_2. We then create 2 virtual folders pointing to folder_1 and folder_2 respectively and mounted them to single user account and have given root permission. Now while moving/renaming files from one virtual folder to another, getting permission denied exception. I am attaching the logs here. Please have a look.

{"level":"info","time":"2022-08-05T11:40:23.790","sender":"HTTP","connection_id":"HTTP_cbmg39q31tdol5a76hd0","message":"rename \"/8digitphonecounts/abcd.txt\"->\"/busenh/abcd.txt\" is not allowed: the paths must be local or on the same virtual folder"}
{"level":"debug","time":"2022-08-05T11:40:23.790","sender":"HTTP","connection_id":"HTTP_cbmg39q31tdol5a76hd0","message":"connection removed, local address \"10.XXX.XX.XXX:8080\", remote address \"10.XX.XXX.XXX:62352\" close fs error: <nil>, num open connections: 3"}

**Comment by drakkan** [comment]
Hi,

as you can see from the logs renames are currently supported between local virtual folders or within the same virtual folder.
As a future improvement, I could allow renaming between virtual folders backed by the same storage provider so the use case above should work.

Renaming between different storage providers requires to first download the file locally and then upload it, this will be never supported: SFTP clients expect a rename response in a timely manner and we have no way to provide a progress for renames.

Please also note that renaming "folders" is not supported for cloud storage providers, take a look at the comment [here](https://github.com/drakkan/sftpgo/blob/890dde0e00d3db6ccf25a6c803d0833c24162dca/internal/vfs/gcsfs.go#L212)

**Reply by rashmimaharana** [reply]
Thanks @drakkan for your response and considering this for future improvement. It will be really nice if you can share me the release tentative date for this feature :)

**Comment by drakkan** [comment]
https://github.com/drakkan/sftpgo/commit/ced4206c5f5e8d9ead18a236267f1b734f4a247f
